bulk-import-tools/internal/orchcli/orchcli.go

```diff
@@ -243,6 +246,56 @@ func (oC *OrchCli) InstanceExists(ctx context.Context, sn, uuid string) (bool, e
 	return false, nil
 }
 
+func (oC *OrchCli) GetHostID(ctx context.Context, sn, uuid string) (string, error) {
+	if uuid == "" {
+		return "", e.NewCustomError(e.ErrInternal)
+	}
+
+	uParsed := *oC.SvcURL
+	uParsed.Path = path.Join(uParsed.Path, fmt.Sprintf("/v1/projects/%s/compute/hosts", oC.Project))
+	query := uParsed.Query()
+	query.Set("filter", fmt.Sprintf("%s=%q", "uuid", uuid))
+	uParsed.RawQuery = query.Encode()
+
+	resp, err := oC.doRequest(ctx, uParsed.String(), http.MethodGet, http.NoBody)
+	if err != nil {
+		return "", e.NewCustomError(e.ErrInternal)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return "", e.NewCustomError(e.ErrInternal)
+	}
+
+	var hosts api.HostsList
+	err = json.NewDecoder(resp.Body).Decode(&hosts)
+	if err != nil {
+		return "", e.NewCustomError(e.ErrInternal)
+	}
+
+	host, err := findAvailableHost(&hosts, sn, uuid)
```

Comment:
With the above filter, we should be able to remove this, and in case we get more than 1 totalElement, we can simply error.

Reply:
Addressed in the other comment.

Reply:
https://github.com/open-edge-platform/infra-core/pull/97#discussion_r2073754190